1-Draft/RFCNNNN-Multi-Line-Continuation.md

```diff
@@ -0,0 +1,301 @@
+
+    As a script/module author,
+    I can wrap commands across multiple lines easily and intuitively without backticks or splatting
+    so that my scripts remain easy to write and maintain while still giving me the benefits of Intellisense and tab completion.
+
+## Specification
+
+* expand the command parser to accept multi-line commands after an at symbol
+(`@`) is encountered at the end of a line
+* terminate multi-line commands when the parser encounters two newlines
+(rather than one), or when the parser encounters any other command-terminating
+token
+
+Note:
+* for commands that do not use the stop-parsing sigil in their arguments,
+command-terminating tokens include a pipe symbol, a redirection operator, a
+closing enclosure, a semi-colon, or a `&` background operator.
+* for commands that do use the stop-parsing sigil in their arguments,
+command-terminating tokens include a pipe symbol or a redirection operator.
+
+## Alternate Proposals and Considerations
+
+### A different sigil
+
+The original draft of this RFC included different options for the sigil that
+could be used to enter multi-line parameter/argument parsing mode, and others
+were presented in the discussion however none of the other sigils that were
+presented could be used without breaking changes. When considering an alternate
+sigil, it must be something that can be identified as a unique token without
+breaking commands that accept multiple strings as positional parameters, such
+as Write-Host (which can write many sigils to the console) or commands external
+to PowerShell.
+
+### Enclosures instead of a sigil
+
+Instead of only requiring a single leading sigil, some users prefer the notion
+of enclosures such that the multi-line command parsing mode would have a very
+clear and well defined start and end. To meet that need, we could follow the
+here-string syntax in PowerShell as an example, offering syntax like the
+following:
+
+```PowerShell
+. {"./plink.exe" @`
+    --% 
+    $Hostname
+    -l $Username
+    -pw $Password
+    $Command
+`@}
+```
+
+All this would do is prevent newline tokens within the enclosures from being
+treated as statement terminators in the current statement.
+
+The closing closure could also be a recognized statement terminator even when
+used after the stop parsing sigil, allowing those commands to be wrapped across
+multiple lines as well.
+
+Like here-string enclosures, we could require that the opening closure be the
+last token on a line. Unlike here-string enclosures, however, it would be
+preferable if the closing closure did not have to be at the start of a line,
+since there is no need for it to be. It would simply have to be the first token
+on line to close the statement, allowing for indentation within scripts.
+
+This alternative also allows for blank lines to be used within the enclosures,
+such that Scenario 3 in @dragonwolf83's comment could be supported and written
+like this:
+
+```PowerShell
+New-ADUser @`
```

Comment:
This is a breaking change because it changes what the trailing backtick means today. Users could absolutely be splitting lines after `@` in the real-world.

Reply:
This is not breaking. Here is output from preview 6:

![image](https://user-images.githubusercontent.com/3515485/70483596-6394e700-1abf-11ea-95c4-f40b4ce6452b.png)

Windows PowerShell 5.1 recognizes it as a parser error as well, and since it's a parser error, this is non-breaking as things stand in PowerShell today.

Also, this doesn't change the meaning of the backtick (a single `` ` `` token). That behavior remains the same. The portion of this RFC that proposes using enclosures suggests we add new enclosure tokens (`` @` `` and `` `@ ``).